### PR TITLE
Enforce jar packaging by default in BuildMojo

### LINKED ISSUE #10807: Improve errors on attempt to run a Quarkus application packaged as a WAR type
**Description**
Following up on user's troubles reported on #10789

Could we raise a nicer, more explicit error message? Question for @stuartwdouglas / @aloubyansky

### PR DESCRIPTION
**Pull request by aloubyansky**
Fixes #10807

### CODE REVIEW

File: devtools/maven/src/main/java/io/quarkus/maven/BuildMojo.java [hunk 1]
```diff
@@ -129,20 +129,28 @@
...
+            return;
+        }
         if (project.getPackaging().equals("pom")) {
             getLog().info("Type of the artifact is POM, skipping build goal");
             return;
         }
-        if (skip) {
-            getLog().info("Skipping Quarkus build");
-            return;
+        if (enforceJarExtension && !project.getArtifact().getArtifactHandler().getExtension().equals("jar")) {
```
Comment (gastaldi) [inline devtools/maven/src/main/java/io/quarkus/maven/BuildMojo.java:150]: Do we support building the project as a WAR?

Comment (aloubyansky) [reply devtools/maven/src/main/java/io/quarkus/maven/BuildMojo.java:150]: I don't think so.

Comment (gastaldi) [reply devtools/maven/src/main/java/io/quarkus/maven/BuildMojo.java:150]: It seems to me that we might not need a flag but just fail the build if the project isn't packaged as a jar?

Comment (stuartwdouglas) [reply devtools/maven/src/main/java/io/quarkus/maven/BuildMojo.java:150]: I also don't think we need the flag.

Comment (geoand) [reply devtools/maven/src/main/java/io/quarkus/maven/BuildMojo.java:150]: +1 for no flag

Comment (Sanne) [reply devtools/maven/src/main/java/io/quarkus/maven/BuildMojo.java:150]: +1
Just fail hard and make sure the error is clear and visible.

Comment (Sanne) [reply devtools/maven/src/main/java/io/quarkus/maven/BuildMojo.java:150]: BTW I suppose you added the flag out of fear of breaking existing applications?

Fear not, it already doesn't work if try to build it as a war. The only problem currently is that the error is very obscure.

Comment (aloubyansky) [reply devtools/maven/src/main/java/io/quarkus/maven/BuildMojo.java:150]: Right, I thought I would add a flag to re-enable the previous behavior (which costs nothing) that could be removed in the next version if nobody complained. I didn't specifically think about war packaging/extensions. People do use custom packaging and extensions out there. But, sure, I'll obey :)

**Review by gastaldi — APPROVED** [review]

**Pull request merged**